File: code/back-end/TellYourFriends/TellYourFriends/Models/Entity/Movie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Web;
using TellYourFriends.Models.Entity;

namespace TellYourFriends.Models
{
    public class Movie
    {
        [Key]
        public int Id { get; set; }

        public string Name { get; set; }

        public string Author { get; set; }

        public string Description { get; set; }

        public string Edition { get; set; }

        public string Image { get; set; }

        public double Rating { get; set; }

        [ForeignKey("User")]
        public int UserId { get; set; }

        public User User { get; set; }

        public List<Category> Categories { get; set; }

        public List<Comment> Comments { get; set; }

    }
}

File: code/front-end/src/app/models/movie.ts
import {Category} from './category';
import {Comment} from './comment';
import {User} from './user';

export class Movie {
  id?: number;
  userId?: number;
  name: string;
  description: string;
  edition: string;
  author: string;
  image?: string;
  rating: number;
  user?: User;
  categories?: Array<Category>;
  comments?: Array<Comment>;

  constructor(item: any) {
    if (typeof(item) === 'number') {
      this.id = item;
    } else {
      this.id = item.Id;
      this.userId = item.UserId;
      this.name = item.Name;
      this.description = item.Description;
      this.edition = item.Edition;
      this.author = item.Author;
      this.image = item.Image;
      this.rating = item.Rating;
      this.categories = item.Categories;
      this.user = item.User;
      this.comments = item.Comments;
    }
  }
}
